File: src/p0998/python/solution.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def insertIntoMaxTree(self, root: TreeNode, val: int) -> TreeNode:
        sentinel = TreeNode(0)
        sentinel.right = root
        prev, curr = sentinel, root
        while curr is not None and curr.val > val:
            prev = curr
            curr = curr.right
        prev.right = TreeNode(val)
        prev.right.left = curr
        return sentinel.right

File: src/p0998/cpp/solution.cpp
#include <iostream>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;

    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    TreeNode *insertIntoMaxTree(TreeNode *root, int val) {
        TreeNode sentinel(0); sentinel.right = root;
        TreeNode *prev = &sentinel, *curr = root;
        while (curr && curr->val > val) {
            prev = curr; curr = curr->right;
        }
        prev->right = new TreeNode(val);
        prev->right->left = curr;
        return sentinel.right;
    }
};
